src/class-post-republisher.php

```diff
@@ -0,0 +1,329 @@
+	 *
+	 * @return void
+	 */
+	public function republish_after_rest_api_request( $post_data ) {
+		$this->republish_request( $post_data->ID, $post_data );
+	}
+
+	/**
+	 * Republishes the original post with the passed post, when using the Classic Editor.
+	 *
+	 * Runs also in the Block Editor to save the custom meta data only when there
+	 * are custom meta boxes.
+	 *
+	 * @param int      $post_id   The copy's post ID.
+	 * @param \WP_Post $post_data The copy's post object.
+	 *
+	 * @return void
+	 */
+	public function republish_after_post_request( $post_id, $post_data ) {
+		if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
+			return;
+		}
+
+		$this->republish_request( $post_id, $post_data );
+	}
+
+	/**
+	 * Republishes the post elements overwriting the original post.
+	 *
+	 * @param \WP_Post $post             The post object.
+	 * @param int      $original_post_id The original post's ID.
+	 *
+	 * @return void
+	 */
+	protected function republish_post_elements( $post, $original_post_id ) {
+		// Cast to array and not alter the original object.
+		$post_to_be_rewritten = (array) $post;
+		// Prepare post data for republishing.
+		$post_to_be_rewritten['ID']          = $original_post_id;
+		$post_to_be_rewritten['post_name']   = \get_post_field( 'post_name', $original_post_id );
+		$post_to_be_rewritten['post_status'] = 'publish';
+
+		// Yoast SEO and other plugins prevent from accidentally updating another
+		// post's data (e.g. the Yoast SEO metadata) by checking the $_POST data
+		// ID with the post object ID. We need to overwrite the $_POST data ID
+		// to allow updating the original post.
+		$_POST['ID'] = $original_post_id;
+		// Republish the original post.
+		$rewritten_post_id = \wp_update_post( \wp_slash( $post_to_be_rewritten ) );
+
+		if ( 0 === $rewritten_post_id ) {
+			\wp_die( \esc_html__( 'An error occurred while republishing the post.', 'duplicate-post' ) );
+		}
+	}
+
+	/**
+	 * Republishes the post taxonomies overwriting the ones of the original post.
+	 *
+	 * @param int      $post_id   The copy's post ID.
+	 * @param \WP_Post $post_data The copy's post object.
+	 *
+	 * @return void
+	 */
+	protected function republish_post_taxonomies( $post_id, $post_data ) {
+		$original_post_id = Utils::get_original_post_id( $post_id );
+
+		$copy_taxonomies_options = [
+			'taxonomies_excludelist' => [],
+			'use_filters'            => false,
+			'copy_format'            => true,
+		];
+		$this->post_duplicator->copy_post_taxonomies( $original_post_id, $post_data, $copy_taxonomies_options );
+	}
+
+	/**
+	 * Republishes the post meta overwriting the ones of the original post.
+	 *
+	 * @param int      $post_id   The copy's post ID.
+	 * @param \WP_Post $post_data The copy's post object.
+	 *
+	 * @return void
+	 */
+	protected function republish_post_meta( $post_id, $post_data ) {
+		$original_post_id = Utils::get_original_post_id( $post_id );
+
+		$copy_meta_options = [
+			'meta_excludelist' => [
```

Comment:
I'm thinking about a single point of storage of these recurring field names to be filtered out by default...
Not for this issue, but as a later refinement